version_check.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+import sys
+from typing import Optional, Dict
+import tempfile
+
+import mistune
+import CppHeaderParser
+import semver
+import git
+
+
+def die(*args, error_code=1, **kwargs):
+    print('ERROR: ', end='', file=sys.stderr)
+    print(*args, **kwargs, file=sys.stderr)
+    exit(error_code)
+
+
+class SemVerWithVPrefix(semver.VersionInfo):
+    @classmethod
+    def parse(cls, version):
+        if not version[0] in ('v', 'V'):
+            raise ValueError(f"{version}: not a valid semantic version tag. Must start with 'v' or 'V'")
+        self = super(SemVerWithVPrefix, cls).parse(version[1:])
+        return self
+
+    def __str__(self):
+        return 'v' + super(SemVerWithVPrefix, self).__str__()
+
+
+def get_header_defines(header_contents: str) -> Dict[str, str]:
+    header = CppHeaderParser.CppHeader(header_contents, argType='string')
+    defines_dict = {}
+    for d in header.defines_detail:
+        define_tup = d['value'].split(' ', maxsplit=1)
+        defines_dict[define_tup[0]] = define_tup[1]
+    return defines_dict
```

Comment:
can be replaced with 
`return dict(map(lambda d: d['value'].split(' ', maxsplit=1), header.defines_detail))`